### PR TITLE
[K8SPSMDB-609] Store additional S3 settings into PBM config

### PR DESCRIPTION
**Pull request by dalbani**
[K8SPSMDB-609](https://jira.percona.com/browse/K8SPSMDB-609) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

https://jira.percona.com/browse/K8SPSMDB-609

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/831/commits/018756e4186e9739a182bdc98601efd2398a8fcd
image: `perconalab/percona-server-mongodb-operator:PR-831-018756e4`

**Review by hors — CHANGES_REQUESTED** [review]

**Comment by defbin** [comment]
LGTM

File: pkg/apis/psmdb/v1/psmdb_types.go [hunk 1]
```diff
@@ -531,6 +531,9 @@ type BackupStorageS3Spec struct {
 	Region            string `json:"region,omitempty"`
 	EndpointURL       string `json:"endpointUrl,omitempty"`
 	CredentialsSecret string `json:"credentialsSecret"`
+	UploadPartSize    int    `json:"uploadPartSize,omitempty"`
+	MaxUploadParts    int    `json:"maxUploadParts,omitempty"`
+	StorageClass      string `json:"storageClass,omitempty"`
```
[the remark this answers is not in the document]
Comment (dalbani) [reply pkg/apis/psmdb/v1/psmdb_types.go:536]: I have just included an example in `deploy/cr.yaml`, based on the default values.
Is that what you had in mind?

**Comment by dalbani** [comment]
Rebased on `main`.

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/831/commits/484c390b4e38a2837ab229a4af02aa055617429d
image: `perconalab/percona-server-mongodb-operator:PR-831-484c390b`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/831/commits/479e2f7be239b30cca749a2597e7a7fb4617b393
image: `perconalab/percona-server-mongodb-operator:PR-831-479e2f7b`

**Review by delgod — APPROVED** [review]

**Review by hors — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| limits | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| scaling | passed |
| arbiter | passed |
| security-context | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| scheduled-backup | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| liveness | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| pitr | failed |
| smart-update | passed |
| version-service | passed |
| users | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/831/commits/8580a99df5815b67aa73fa265ac1facabb8b5a00
image: `perconalab/percona-server-mongodb-operator:PR-831-8580a99d`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| arbiter | passed |
| self-healing-chaos | passed |
| service-per-pod | passed |
| operator-self-healing | passed |
| liveness | passed |
| operator-self-healing-chaos | passed |
| smart-update | passed |
| version-service | passed |
| init-deploy | passed |
| users | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| pitr | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/831/commits/8580a99df5815b67aa73fa265ac1facabb8b5a00
image: `perconalab/percona-server-mongodb-operator:PR-831-8580a99d`

**Review by defbin — APPROVED** [review]

**Comment by dalbani** [comment]
Cool 😎

**Pull request merged**